### PR TITLE
fix isEqual in luxon utils

### PR DESCRIPTION
**Pull request by sakulstra**
- fixes https://github.com/dmtrKovalenko/material-ui-pickers/issues/870

disclaimer: I didn't really look at the pickers code yet. Wondering if it makes more sense to prevent the isEqual(null, sth) check somewhere in the code over there. Anyhow, it shouldn't fail

### CODE REVIEW

File: packages/luxon/src/luxon-utils.ts [hunk 1]
```diff
@@ -67,7 +67,10 @@ export default class LuxonUtils implements IUtils<DateTime> {
       return true;
     }
 
-    return this.date(value).equals(this.date(comparing));
+    const date = this.date(value);
+    if (!date) return false;
```
[the remark this answers is not in the document]
Comment (sakulstra) [reply packages/luxon/src/luxon-utils.ts]: It was more like a quick-fix @endOfWorkingDay kind of fix :D didn't check to carefully

As I get it the root problem is that `this.date(null)` on luxon doesn't return a luxon object so equals can't be called on undefined. This fix simply returns false in case this.date doesn't return a valid luxon date.

Tl;Dr;
It's fixing the production issue, but probably not the cleanest solution

[the remark this answers is not in the document]
Comment (sakulstra) [reply packages/luxon/src/luxon-utils.ts]: @dmtrKovalenko any update on this, anything i can do? This bug currently breaks our sign-up process
